File: supervision-server-data/src/main/java/supervision/server/reporting/command/LongSearchCriteria.java
package supervision.server.reporting.command;

import lombok.Data;
import lombok.NonNull;

/**
 * {@code @Data} annotation creates getter and setter.
 * A constructor with all fields set as {@code @NonNull} will also be created
 * 
 * @author cyrilg
 *
 */
@Data
public class LongSearchCriteria implements SearchCriteria {
	
	@NonNull
	private String key;
	@NonNull
	private ReportOperator operator;
	@NonNull
	private Long value;
	
}

File: supervision-client/src/app/shared/technical/reporting/long-search-criteria.ts
import { SearchCriteria } from "./search-criteria";
import { ReportOperator } from "./report-operator";

export class LongSearchCriteria implements SearchCriteria {
    key: string;
    operator: ReportOperator;
    value: number;

    criteriaType: string = 'long';

    public constructor(key: string, operator: ReportOperator, value: number) {
        this.key = key;
        this.operator = operator;
        this.value = value;
    }

    getValue(): number {
        return this.value;
    }
}